code/harness/DiskContents.cpp

```diff
@@ -0,0 +1,298 @@
+#include "DiskContents.h"
+using std::endl;
+
+namespace fs_testing {
+
+fileAttributes::fileAttributes() {
+  dir_attr = NULL;
+  stat_attr = NULL;
+  md5sum = "";
+}
+
+fileAttributes::~fileAttributes() {
+}
+
+void fileAttributes::set_dir_attr(struct dirent* a) {
+  dir_attr = (struct dirent *) malloc(sizeof(struct dirent));
+  dir_attr->d_ino = a->d_ino;
+  dir_attr->d_off = a->d_off;
+  dir_attr->d_reclen = a->d_reclen;
+  dir_attr->d_type = a->d_type;
+  strncpy(dir_attr->d_name, a->d_name, sizeof(a->d_name));
+  dir_attr->d_name[sizeof(a->d_name) - 1] = '\0';
+}
+
+void fileAttributes::set_stat_attr(struct stat* a) {
+  stat_attr = (struct stat *) malloc(sizeof(struct stat));
```

Comment:
if you are copying all the fields of the struct I think you can just use the `=` operator